gensim/utils.py

```diff
@@ -139,6 +140,36 @@ def file_or_filename(input):
         return input
 
 
+@contextmanager
+def open_file(input):
+    """ Generates 'with' like behaviour except closing the file object
+
+    Parameters
+    ----------
+    input : str or file-like
+        Filename or file-like object.
+
+    Yields
+    -------
+    str
+        lines from the given file-like object or a new file-like object from the given filename.
+
+    """
+    mgr = file_or_filename(input)
+    exc = False
+    try:
+        yield mgr
+    except Exception:
+        # Handling any unhandled exceptions from the code nested in 'with' statement.
+        exc = True
+        if not mgr.__exit__(*sys.exc_info()):
```

Comment:
what return `__exit__` ?

Reply:
Sorry , I didn't understand your query.

Reply:
`X = mgr.__exit__(*sys.exc_info())` , what is `X` in this case?

Reply:
x is True if the exception was handled gracefully , else False